### ISSUE TITLE
DB/Quest: Fei Fei's Treat

### ISSUE THREAD

**Issue by Trista** [TrinityCore/TrinityCore#8584]
This is from a stuff Bio wanted to be converted from dropped quest_start_scripts to SAI, this is the converted and make using some proper spells/mechanic version of the quest.
However, when I finished it I realised it never was correct, in db there are a lot of bunny spawns. Seems they are used to move to each of them and finally somehow when reached last one, should cast the spell.
Any info, idea about this?
MOVE_TO maybe can be used nested multiple times within an actionlist that will target the closest bunny in a pair with despawn closest of that entry.
But if it is like this is pretty hard to follow and a lot of core bugs may need a fix first, anyway won't be around for some time, so any suggestions are welcomed. Maybe the creatore of the orignal quest_start_script will have some ideas what is for what. The other way would be that in blizzard the mechanic is bugged and stash would have random pop /but I doubt that/.
```sql
-- Quest {A}: Fei Fei's Treat - http://www.wowhead.com/quest=10919
-- ID indexes
SET @NPC_FEI_FEI := 24068;
SET @NPC_FEI_FEI_STASH_BUNNY := 22467;
SET @WP_FEI_FEI := @NPC_FEI_FEI * 10;
SET @SPELL_FEI_FEI_STASH := 39216; -- Create Fei Fei Stash

-- Waypoints - Fei Fei waypoints on the way to his stash
DELETE FROM `waypoints` WHERE `entry`=@WP_FEI_FEI;
INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
(@WP_FEI_FEI,1,-697.934,2612.59,89.4752, 'Fei Fei - Point 1'),
(@WP_FEI_FEI,2,-688.665,2580.78,86.9841, 'Fei Fei - Point 2'),
(@WP_FEI_FEI,3,-689.199,2588.81,87.446, 'Fei Fei - Point 3'),
(@WP_FEI_FEI,4,-667.044,2611.74,85.7029, 'Fei Fei - Point 4'),
(@WP_FEI_FEI,5,-649.407,2636.82,86.1539, 'Fei Fei - Point 5'),
(@WP_FEI_FEI,6,-659.047,2651.34,87.048, 'Fei Fei - Point 6'),
(@WP_FEI_FEI,7,-656.048,2652.06,86.5892, 'Fei Fei - Point 7'),
(@WP_FEI_FEI,8,-656.055,2674.2,88.1354, 'Fei Fei - Point 8'),
(@WP_FEI_FEI,9,-691.042,2652.45,92.1508, 'Fei Fei - Point 9'),
(@WP_FEI_FEI,10,-688.807,2701.25,94.8354, 'Fei Fei - Point 10'),
(@WP_FEI_FEI,11,-667.514,2716.08,94.4471, 'Fei Fei - Point 11'),
(@WP_FEI_FEI,12,-688.593,2723.67,94.4145, 'Fei Fei - Point 12'),
(@WP_FEI_FEI,13,-687.635,2743.1,93.9095, 'Fei Fei - Point 13'),
(@WP_FEI_FEI,14,-697.062,2748.04,93.938, 'Fei Fei - Point 14'),
(@WP_FEI_FEI,15,-687.635,2743.1,93.9095, 'Fei Fei - Point 15'),
(@WP_FEI_FEI,16,-685.121,2689.01,93.8042, 'Fei Fei - Point 16'),
(@WP_FEI_FEI,17,-688.862,2627.19,89.8591, 'Fei Fei - Point 17'),
(@WP_FEI_FEI,18,-685.494,2626.74,89.2711, 'Fei Fei - Point 18');

-- SAI/template support
UPDATE `creature_template` SET `AIName`='SmartAI' WHERE `entry`=@NPC_FEI_FEI;
DELETE FROM `smart_scripts` WHERE `entryorguid`=@NPC_FEI_FEI AND `source_type`=0;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(@NPC_FEI_FEI,0,0,0,19,0,100,0,10919,0,0,0,53,0,0,0,0,60000,0,1,0,0,0,0,0,0,0,'Fei Fei - On quest 10919 accepted by player - Start waypoint movement (On finish despawn after 1 min'),
(@NPC_FEI_FEI,0,1,0,39,0,100,0,1,@WP_FEI_FEI,0,0,54,3000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 1 started - WP pause'),
(@NPC_FEI_FEI,0,2,0,39,0,100,0,2,@WP_FEI_FEI,0,0,54,10000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 2 started - WP pause'),
(@NPC_FEI_FEI,0,3,0,39,0,100,0,3,@WP_FEI_FEI,0,0,54,24000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 3 started - WP pause'),
(@NPC_FEI_FEI,0,4,0,39,0,100,0,4,@WP_FEI_FEI,0,0,54,27000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 4 started - WP pause'),
(@NPC_FEI_FEI,0,5,0,39,0,100,0,5,@WP_FEI_FEI,0,0,54,35000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 5 started - WP pause'),
(@NPC_FEI_FEI,0,6,0,39,0,100,0,6,@WP_FEI_FEI,0,0,54,42000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 6 started - WP pause'),
(@NPC_FEI_FEI,0,7,0,39,0,100,0,7,@WP_FEI_FEI,0,0,54,52000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 7 started - WP pause'),
(@NPC_FEI_FEI,0,8,0,39,0,100,0,8,@WP_FEI_FEI,0,0,54,54000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 8 started - WP pause'),
(@NPC_FEI_FEI,0,9,0,39,0,100,0,9,@WP_FEI_FEI,0,0,54,68000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 9 started - WP pause'),
(@NPC_FEI_FEI,0,10,0,39,0,100,0,10,@WP_FEI_FEI,0,0,54,80000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 10 started - WP pause'),
(@NPC_FEI_FEI,0,11,0,39,0,100,0,11,@WP_FEI_FEI,0,0,54,93000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 11 started - WP pause'),
(@NPC_FEI_FEI,0,12,0,39,0,100,0,12,@WP_FEI_FEI,0,0,54,103000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 12 started - WP pause'),
(@NPC_FEI_FEI,0,13,0,39,0,100,0,13,@WP_FEI_FEI,0,0,54,108000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 13 started - WP pause'),
(@NPC_FEI_FEI,0,14,0,39,0,100,0,14,@WP_FEI_FEI,0,0,54,115000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 14 started - WP pause'),
(@NPC_FEI_FEI,0,15,0,39,0,100,0,15,@WP_FEI_FEI,0,0,54,125000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 15 started - WP pause'),
(@NPC_FEI_FEI,0,16,0,39,0,100,0,16,@WP_FEI_FEI,0,0,54,130000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 16 started - WP pause'),
(@NPC_FEI_FEI,0,17,0,39,0,100,0,17,@WP_FEI_FEI,0,0,54,145000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 17 started - WP pause'),
(@NPC_FEI_FEI,0,18,0,39,0,100,0,18,@WP_FEI_FEI,0,0,54,165000,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 18 started - WP pause'),
(@NPC_FEI_FEI,0,19,0,39,0,100,0,18,@WP_FEI_FEI,0,0,11,@SPELL_FEI_FEI_STASH,0,0,0,0,0,1,0,0,0,0,0,0,0,'Fei Fei - On WP point 18 reached - Cast Create Fei Fei Stash');

-- Conditions
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=13 AND `SourceEntry`=@SPELL_FEI_FEI_STASH;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES 
(13,1,@SPELL_FEI_FEI_STASH,0,0,31,0,3,@NPC_FEI_FEI_STASH_BUNNY,0,0,0,'', 'Create Fei Fei Stash can target only Fei Fei Stash Dummy');
```

**Comment by Aokromes** [comment]
What's missing here?

**Comment by dr-j** [comment]
This should be closed I recently scripted this again seems missed this 

https://github.com/TrinityCore/TrinityCore/commit/576f37934416262e0d71ceff3635b06efb9e314e

**Comment by tkrokli** [comment]
Thanks for linking the commit, **dr-j**  :+1: I knew I had seen something about it before, but I could not find exactly where it was posted before you commented here. (I almost thought "have I dreamt this?") :smile:
